running-code/execute-script.md

```diff
@@ -0,0 +1,216 @@
+Linux, macOS, all BSDs, and many other operating systems are fully- or mostly-POSIX compliant.
+
+Windows is not natively POSIX compliant. However, some "modes" inside of Windows are, such as [WSL](https://learn.microsoft.com/en-us/windows/wsl/about)
+(Windows Subsystem for Linux), gitbash, or some VSCode terminals.
+:::
+
+### Executing Python scripts on Windows
+
+If your Windows machine has Python registered as the default application associated with `.py` files, then any Python
+scripts can be run as commands. However, only one Python can be registered at a time, so all Python scripts run this
+way will use the same Python environment.
+
+Additionally, most Windows Python installs come with the [Python Launcher](https://docs.python.org/3/using/windows.html#python-launcher-for-windows).
+This launcher allows you to specify the version of Python to run and also can read shebang lines and emulate some of that behavior.
+This allows you to reuse shebang lines between Linux, macOS, and Windows systems. However, on Windows, the command must still
+be prefaced with another command (`py`).
+
+```bash
+py my_program.py
+# ✨ Hello from Python ✨
+```
+
+:::{tip}
+All Python files should end in a `.py`. The `.py` is required for Windows to associate a script with Python. In comparison, on Linux and Mac machines, `.py` is a recommended best practice;
+the shebang (`#!`) associates the script execution with Python.
+
+Windows doesn’t rely on a shebang (`#!`) to determine how to execute a Python file. Instead, Python sees the first `#` and treats the line as a comment and ignores it!
+
+For reproducibility, which ensures that your code runs consistently across machines, you should always include the `.py` extension and the shebang (`#!`) in your Python scripts.
+:::
+
+(execute-script-comparison)=
+## Comparing passing files to Python vs the "shebang" (`#!`) execution approach
+
+There are pros and cons of using the two approaches above to execute Python files.
+
+### Pros of passing a file to `python`:
+- It works with all files: The file path passed to Python doesn't have to end in a `.py` (not even on Windows)
+- It works with more restrictive fiels: The file doesn't need execute permissions
+- It's reproducible: This approach works on every system
+- You are being explicit about what you expect to happen
+
+### Pros of associateing your script with a launcher:
+- The Python version used is assicated with a specific Python installation
+- That Python can be different that the Python installation currenlty active in a [`virtual-environment`](https://packaging.python.org/en/latest/guides/installing-using-pip-and-virtual-environments/#create-and-use-virtual-environments)
+- You don't have to remember when using the command which Python installation the script should be associated with, or even that the script is written in Python
+
+(execute-script-name-eq-main)=
+## Separating script from import behavior
```

Comment:
ya ya as said above i think this works better in the executing packages lesson since importing scripts is super fragile and ime rare